crates/core/src/rpc/minimal.rs

```diff
@@ -754,6 +784,106 @@ impl Minimal for SurfpoolMinimalRpc {
 mod tests {
     use super::*;
     use crate::tests::helpers::TestSetup;
+    use solana_commitment_config::CommitmentConfig;
+    use solana_client::rpc_config::RpcContextConfig;
+    use solana_epoch_info::EpochInfo;
+   
+    #[test]
+    fn test_get_block_height_debug() {
+        let setup = TestSetup::new(SurfpoolMinimalRpc);
+
+        // defaults to finalized, finalized should be 31 lesser than latest height
+        let expected_height = setup.rpc.get_block_height(Some(setup.context.clone()), None);
+        
+        let latest_epoch_block_height = setup.context.svm_locker.with_svm_reader(|svm_reader| {
+            svm_reader.latest_epoch_info.block_height
+        });
+        
+        assert_eq!(latest_epoch_block_height, expected_height.unwrap() + FINALIZATION_SLOT_THRESHOLD);
+    }
+
+    #[test]
+    fn test_get_block_height_processed_commitment() {
+        let setup = TestSetup::new(SurfpoolMinimalRpc);
+ 
+        let config = RpcContextConfig {
+            commitment: Some(CommitmentConfig::processed()),
+            min_context_slot: None,
+        };
+        
+        let expected_height = setup.rpc.get_block_height(Some(setup.context.clone()), Some(config));
+        assert!(expected_height.is_ok());
+        
+        let latest_epoch_block_height = setup.context.svm_locker.with_svm_reader(|svm_reader| {
+            svm_reader.latest_epoch_info.block_height
+        });
+        
+        assert_eq!(expected_height.unwrap(), latest_epoch_block_height);
+    }
+
+    #[test]
+    fn test_get_block_height_confirmed_commitment() {
+        let setup = TestSetup::new(SurfpoolMinimalRpc);
+ 
+        let config = RpcContextConfig {
+            commitment: Some(CommitmentConfig::confirmed()),
+            min_context_slot: None,
+        };
+        
+        let expected_height = setup.rpc.get_block_height(Some(setup.context.clone()), Some(config));
+        assert!(expected_height.is_ok());
+        
+        let latest_epoch_block_height = setup.context.svm_locker.with_svm_reader(|svm_reader| {
+            svm_reader.latest_epoch_info.block_height
+        });
+        
+        assert_eq!(expected_height.unwrap(), latest_epoch_block_height - 1);
+    }
+
+    #[test]
+    fn test_get_block_height_with_min_context_slot() {
```

Comment:
This test still isn't checking the error case, where we hit:
```rust
            if !block_exists {
                return Err(jsonrpc_core::Error::invalid_params(format!(
                    "Block not found for slot: {}", target_slot
                )));
            }
```

Can you add a test for that?